File: custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int> alph(26);
        for(char c : s)
            alph[c-'a']++;
        
        string ans;
        for(char c : order)
            if(alph[c-'a'])
                ans += string(alph[c-'a'], c), alph[c-'a'] = 0;

        for(int i = 0; i < alph.size(); i++)
            if(alph[i])
                ans += string(alph[i], i+'a');

        return ans;
    }
};

File: custom-sort-string.cs
public class Solution {
    public string CustomSortString(string order, string s) {
        int[] alph = new int[26];
        foreach(char c in s)
            alph[c-'a']++;
        
        String ans = "";
        foreach(char c in order){
            int i = c-'a';
            if(alph[i] != 0){
                ans += new String(c, alph[i]);
                alph[i] = 0;
            }
        }

        for(int i = 0; i < alph.Length; i++)
            if(alph[i] != 0)
                ans += new String((char)(i+'a'), alph[i]);

        return ans;
    }
}